File: BlackJackJava/Hand.java
import java.util.ArrayList;
import java.util.List;

public class Hand {
    private List<Card> hand = new ArrayList<>();

    public void addCard(Card card){
        hand.add(card);
    }

    public int calculateValue(){
        int sum = 0; 
        int aceNumber = 0;
        for(Card card : hand){
            String cardValue = card.getValue();
            if(cardValue.equals("J") || cardValue.equals("Q")  || cardValue.equals("K")){
                sum = sum + 10;
            }
            else if(cardValue.equals("A")){
                sum = sum + 11;
                aceNumber = aceNumber + 1;
            }
            else{
                sum = sum + Integer.parseInt(cardValue);
            }
        }

        while(aceNumber > 0 && sum > 21){
            sum = sum - 10;
            aceNumber = aceNumber - 1;
        }
        return sum;
    }

    public int calculateCardValue(Card card){
        String cardValue = card.getValue();
        if(cardValue.equals("J")||cardValue.equals("Q")||cardValue.equals("K")){
            return 10;
        }
        else if(cardValue.equals("A")){
            return 11;
        }
        else{
            return Integer.parseInt(cardValue);
        }
    }

    public List<Card> getCard(){ 
        return hand;
    }
    
    public void resetHand(){
        hand.clear();
    }
    
    public String toString(){
        return hand.toString();
    }

}

File: BlackJavaScript/Hand.js
class Hand{
    constructor(){
        this.hand = [];
    }
    addCard(card){
        this.hand.push(card);
    }
    calculateValue(){
        let sum = 0; 
        let aceCount = 0; 
        for(const card of this.hand){
            const cardValue = card.getValue();
            if(cardValue === "J" || cardValue === "Q" || cardValue === "K"){
                sum += 10;
            }
            else if(cardValue === "A"){
                sum += 11;
                aceCount += 1;
            }
            else{
                sum += parseInt(cardValue, 10);
            }
        }
        while(aceCount > 0 && sum > 21){
            sum -= 10;
            aceCount -= 1;
        }
        return sum;
    }

    calculateCardValue(card){
        const cardValue = card.getValue();
        if(cardValue === "J" || cardValue ===  "Q" || cardValue === "K"){
            return 10;
        }
        else if(cardValue === "A"){
            return 11;
        }
        else{
            return parseInt(cardValue, 10);
        }
    }

    getCard(){
       return this.hand
    }
    restHand(){
        this.hand = [];
    }
    toString(){
        this.hand.map(card => card.toString()).join(', ');
    }
}